docs/atomic-swap-protocol-v1.md

```diff
@@ -0,0 +1,156 @@
+to use a sufficiently good source of entropy to generate it, but this is not
+required by the protocol.
+
+The hashing algorithm is fixed to SHA-256. The hash is 32 bytes of raw binary
+data with `hash = sha256(preimage)`.
+
+### Timeouts
+
+TODO
+
+### Swap offer actions
+
+Actions are mutations of the on-chain state triggered by transactions.
+
+- **Create:** By creating a swap offer, the funds are locked in the contract. A
+  timeout and a hash must be specified.
+- **Claim:** By revealing a preimage for the hash before the timeout is reached,
+  the contract transfers the tokens to the recipient. Requires a preimage of
+  exactly 32 bytes. Common synonyms: _release_, _execute_.
+- **Abort:** Withdraws the tokens from the contract to the creator. Requires
+  timeout to be reached.
+
+### Swap offer open state
+
+Each swap offer individually is either _open_, _claimed_ or _aborted_. The open
+state can only be changed by performing an action via a transaction.
+
+- **Open:** This is the initial open state.
+- **Claimed:** An _open_ swap becomes _claimed_ by performing the clain action.
+- **Aborted:** An _open_ swap becomes _aborted_ by performing the abort action.
+
+Common synonym: _settled = (claimed or aborted)_
+
+### Swap offer expiry state
+
+Each swap offer individually is either _non-expired_ or _expired_. The state
+automatically changes on-chain as the underlying time (wall time or block
+height) increases. A state can switch from _non-expired_ to _expired_ but not
+vice versa.
+
+- **Non-expired:** In the range `[creation, timeout)`, the swap offer is
+  _non-expired_. This is the default state for most offers. However, an offer
+  can be created in _expired_ state directly if the above interval is empty.
+- **Expired** In the range `[timeout, ∞)`, the swap offer is _expired_. Common
+  synonym: _abortable_.
+
+#### Note on open and expiry state
+
+Both states are independent of each other. A swap offer can be open and expired
+at the same time.
+
+## Risks
+
+The following risks for cross-chain atomic swaps have been identified. Some of
+them are fully avoided by this protocol but still documented for tranparancy.
+Others need to be mitigated by smart choice of parameters.
+
+### Pre-image attacks
+
+The short position copies the hash without knowing the preimage. This means the
+long position could use a very long preimage that is either expensive or
+impossible to process on one chain but works easily on the other chain. This can
+make it unprofitable or impossible for short to claim their tokens, leading to
+free tokens for long.
+
+This risk is avoided in this protocol by having a fixed preimage length.
+
+### No counter offer created
+
+The short position can pretend to have the intention to create a counter offer
+but not do it. In this case long loses access to the locked tokens until the
+timeout is reached.
+
+### Half executed atomic swaps
+
+TODO: timeout issue etc.
```

Comment:
Add here text about choice of timeout.

Main point is the short position must expire before the long position.
And with a good margin of time (eg. 24 hours), so it is impossible that the short position is claimed eg. 1 minute before expiry and the counter-party sleeps through their window to claim the other half.

This is on the end users to choose/ensure proper time envelopes, but this should be well explained.